File: Groupe1/TP1/src/boucles.py
# Demande à l'utilisateur d'entrer la taille du triangle et la stocke dans la variable 'taille'
taille = int(input("Quelle est la taille de votre triangle ?"))

# Définition de la fonction triangle_rectangle qui dessine un triangle rectangle avec la taille donnée
def triangle_rectangle(taille):

    # Vérification si la taille dépasse 10
    if taille > 10:
        # Si la taille est supérieure à 10, un message est affiché et la fonction s'arrête
        print ("La taille du triangle doit être inférieur à 10 pour avoir un triangle rectangle")
        return # Fin prématurée de la fonction

    # Boucle pour générer chaque ligne du triangle
    for i in range(1,taille+1): # La boucle parcourt chaque ligne de 1 jusqu'à 'taille'
        ligne=""    # Initialisation de la chaîne de caractères qui va représenter la ligne

        # Si on est sur la première ou la dernière ligne
        if i==1 or i==taille:

            # On remplit la ligne avec 'i' étoiles (une étoile par unité de hauteur)
            ligne+=i*"*"
        else:
            # Pour les lignes intermédiaires
            ligne="*" # On commence toujours par une étoile
            
            # Boucle pour insérer des '#' dans les positions intermédiaires
            
            for j in range (1,i-1): # Ajout de 'i-2' caractères '#' (dépend de la ligne actuelle)
                ligne+="#"
            
            # On termine la ligne avec une autre étoile
            ligne+="*"

        # Affichage de la ligne générée
        print(ligne)


# Appel de la fonction pour dessiner le triangle avec la taille fournie par l'utilisateur

triangle_rectangle(taille)

File: Groupe1/TP1/src/boucles.c
#include <stdio.h> // Inclusion de la bibliothèque standard pour les fonctions d'entrée/sortie (printf, scanf).

// Fonction qui dessine un triangle rectangle avec une taille donnée
void triangle_rectangle(int taille) {

    // Vérification si la taille dépasse 10
    if (taille > 10) {
        // Si la taille est supérieure à 10, un message est affiché et la fonction retourne
        printf("La taille du triangle doit être inférieure à 10 pour avoir un triangle rectangle\n");
        return;// On quitte la fonction sans dessiner le triangle
    }
    // Boucle pour créer chaque ligne du triangle
    for (int i = 1; i <= taille; i++) {
        // Si on est à la première ligne ou à la dernière, on dessine seulement des '*'
        if (i == 1 || i == taille) {
            // Boucle pour afficher 'i' étoiles
            for (int j = 0; j < i; j++) {
                printf("*"); // Affiche une étoile
            }
        } else {
            // Si ce n'est pas la première ou la dernière ligne
            printf("*"); // On commence la ligne par une étoile
            // Boucle pour afficher des '#' entre les étoiles, sauf si on est sur la 2e ligne
            for (int j = 1; j < i - 1; j++) {
                printf("#"); // Affiche des '#'
            }
            printf("*"); // On termine la ligne par une étoile
        }
        // Passage à la ligne suivante après chaque ligne du triangle
        printf("\n");
    }
}

int main() {
    int taille; // Déclaration d'une variable pour stocker la taille du triangle
    // Demande à l'utilisateur de saisir la taille du triangle
    printf("Quelle est la taille de votre triangle ? ");
    scanf("%d", &taille); // Lecture de la taille entrée par l'utilisateur
    // Appel de la fonction pour dessiner le triangle rectangle
    triangle_rectangle(taille);

    return 0; // Fin du programme
}
